### PR TITLE
Feature/issue-265 

### PR DESCRIPTION
**Pull request by moranthomas**

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -54,6 +54,13 @@ function setIfValue(target, path, value) {
   set(target, path, value)
 }
 
+function uniqueOnly(a) {
```
Comment (AlexWebYourmind) [inline lib/utils.js]: @moranthomas I am a little bit confused by the method/param naming.
I do understand that this method is supposed to deduplicate the data somehow based on properties, but I am under the impression we should use a more eloquent naming and adhere to general JS best practices: "**A function name should be a verb or a verb phrase, and it needs to communicate its intent, as well as the order and intent of the arguments**."

Would like also @storrisi to share his comments.

Other than that, I am happy with the implementation.

File: routes/originNpm.js [hunk 2]
```diff
@@ -4,6 +4,7 @@
 const asyncMiddleware = require('../middleware/asyncMiddleware')
 const router = require('express').Router()
 const requestPromise = require('request-promise-native')
+const utils = require('../lib/utils')
```
Comment (storrisi) [inline routes/originNpm.js]: I don't see the implementation applied here, is that correct?

File: routes/originNuget.js [hunk 3]
```diff
@@ -4,6 +4,7 @@
 const asyncMiddleware = require('../middleware/asyncMiddleware')
 const router = require('express').Router()
 const requestPromise = require('request-promise-native')
+const utils = require('../lib/utils')
```
Comment (storrisi) [inline routes/originNuget.js]: Same here, it doesn't need to check the uniqueness of the results?

**Comment by storrisi** [comment]
Given that all the origin routes returns a list, i don't understand why only some of them had applied the uniqueness.
It should be a default method, the website would never receive duplicate data.

I've also suggested to use lodash instead of a custom method, to keep consistency with the codebase. Everywhere else has been preferred to using a lib where it could be avoided to write a custom method.

File: lib/utils.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (moranthomas) [reply lib/utils.js]: I looked at lodash but have seen comments that it is not optimized for performance https://stackoverflow.com/questions/9229645/remove-duplicate-values-from-js-array

[the remark this answers is not in the document]
Comment (moranthomas) [reply lib/utils.js]: Probably performance is not an issue for this small data set use case but that was the justification for going with the custom function

Comment (moranthomas) [reply lib/utils.js]: Agreed, the function and it's params could be more clear in their intent by naming convention. Will implement.

File: routes/originMaven.js [hunk 4]
```diff
@@ -14,7 +15,8 @@ router.get(
       const url = `https://search.maven.org/solrsearch/select?q=g:"${group}"+AND+a:"${artifact}"&core=gav&rows=100&wt=json`
       const answer = await requestPromise({ url, method: 'GET', json: true })
       const result = answer.response.docs.map(item => item.v)
-      return response.status(200).send(result)
+      const unique = utils.uniqueOnly(result)
+      return response.status(200).send(unique)
```
[the remark this answers is not in the document]
Comment (moranthomas) [reply routes/originMaven.js]: Agreed. Will implement in next commit

File: lib/utils.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlexWebYourmind) [reply lib/utils.js]: I think we can stick to Lodash right now, performance gain will be close to null for such a small data set.

File: lib/utils.js [hunk 5]
```diff
@@ -209,5 +209,5 @@ module.exports = {
   updateSourceLocation,
   isLicenseFile,
   normalizeSpdx,
-  uniqueOnly
+  filterForUniqueItemsOnly
```
Comment (AlexWebYourmind) [inline lib/utils.js]: @moranthomas Better, even tho I think the extra "Only" in the naming might be not necessary. No worries about that for now.

File: routes/originNpm.js (hunk 2 above)
Comment (moranthomas) [reply routes/originNpm.js]: Have included implementation here now also

File: routes/originNuget.js (hunk 3 above)
Comment (moranthomas) [reply routes/originNuget.js]: Have included implementation here now also

**Pull request merged**